src/lightning/data/cache/serializers.py

```diff
@@ -0,0 +1,223 @@
+        return ints.tobytes() + mode + raw, None
+
+    def deserialize(self, data: bytes) -> Any:
+        idx = 3 * 4
+        width, height, mode_size = np.frombuffer(data[:idx], np.uint32)
+        idx2 = idx + mode_size
+        mode = data[idx:idx2].decode("utf-8")
+        size = width, height
+        raw = data[idx2:]
+        return Image.frombytes(mode, size, raw)  # pyright: ignore
+
+    def can_serialize(self, item: Any) -> bool:
+        return isinstance(item, Image.Image) and not isinstance(item, JpegImageFile)
+
+
+class IntSerializer(Serializer):
+    """The IntSerializer serialize and deserialize integer to and from bytes."""
+
+    def serialize(self, item: int) -> Tuple[bytes, Optional[str]]:
+        return str(item).encode("utf-8"), None
+
+    def deserialize(self, data: bytes) -> int:
+        return int(data.decode("utf-8"))
+
+    def can_serialize(self, item: Any) -> bool:
+        return isinstance(item, int)
+
+
+class JPEGSerializer(Serializer):
+    """The JPEGSerializer serialize and deserialize JPEG image to and from bytes."""
+
+    def serialize(self, item: Image) -> Tuple[bytes, Optional[str]]:
+        if isinstance(item, JpegImageFile):
+            if not hasattr(item, "filename"):
+                raise ValueError(
+                    "The JPEG Image's filename isn't defined. HINT: Open the image in your Dataset __getitem__ method."
+                )
+            with open(item.filename, "rb") as f:
+                return f.read(), None
+        raise TypeError(f"The provided itemect should be of type {JpegImageFile}. Found {item}.")
+
+    def deserialize(self, data: bytes) -> Union[JpegImageFile, torch.Tensor]:
+        if _TORCH_VISION_AVAILABLE:
+            array = torch.frombuffer(data, dtype=torch.uint8)
+            return decode_jpeg(array)
+
+        inp = BytesIO(data)
+        return Image.open(inp)
+
+    def can_serialize(self, item: Any) -> bool:
+        return isinstance(item, JpegImageFile)
+
+
+class BytesSerializer(Serializer):
+    """The BytesSerializer serialize and deserialize integer to and from bytes."""
+
+    def serialize(self, item: bytes) -> Tuple[bytes, Optional[str]]:
+        return item, None
+
+    def deserialize(self, item: bytes) -> bytes:
+        return item
+
+    def can_serialize(self, item: bytes) -> bool:
+        return isinstance(item, bytes)
+
+
+_TORCH_DTYPES_MAPPING = {
+    0: torch.float32,
+    1: torch.float,
+    2: torch.float64,
+    3: torch.double,
+    4: torch.complex64,
+    5: torch.cfloat,
+    6: torch.complex128,
+    7: torch.cdouble,
+    8: torch.float16,
+    9: torch.half,
+    10: torch.bfloat16,  # Not supported https://github.com/pytorch/pytorch/issues/110285
+    11: torch.uint8,
+    12: torch.int8,
+    13: torch.int16,
+    14: torch.short,
+    15: torch.int32,
+    16: torch.int,
+    17: torch.int64,
+    18: torch.long,
+    19: torch.bool,
+}
+
+
+class TensorSerializer(Serializer):
```

Comment:
Have you tested this with tensor subclasses such as nested tensors or distributed tensors?

Reply:
I removed support for subclass for now.